src/source.ts

```diff
@@ -32,13 +33,30 @@ export default class Source extends React.Component<Props, void> {
 
     if (!map.getSource(this.id) && (geoJsonSource || tileJsonSource)) {
       map.addSource(this.id, (geoJsonSource || tileJsonSource) as any);
+      map.on('load', this.onData);
```

Comment:
don't you just want to listen the load of a single source only ? (the one just above)